server/src/main/java/org/elasticsearch/action/admin/indices/mapping/get/GetFieldMappingsResponse.java

```diff
@@ -112,26 +114,49 @@ public FieldMappingMetaData fieldMappings(String index, String type, String fiel
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        boolean includeTypeName = params.paramAsBoolean(BaseRestHandler.INCLUDE_TYPE_NAME_PARAMETER, true);
+
         builder.startObject();
         for (Map.Entry<String, Map<String, Map<String, FieldMappingMetaData>>> indexEntry : mappings.entrySet()) {
             builder.startObject(indexEntry.getKey());
             builder.startObject(MAPPINGS.getPreferredName());
-            for (Map.Entry<String, Map<String, FieldMappingMetaData>> typeEntry : indexEntry.getValue().entrySet()) {
-                builder.startObject(typeEntry.getKey());
-                for (Map.Entry<String, FieldMappingMetaData> fieldEntry : typeEntry.getValue().entrySet()) {
-                    builder.startObject(fieldEntry.getKey());
-                    fieldEntry.getValue().toXContent(builder, params);
+
+            if (includeTypeName == false) {
```

Comment:
Sure, covering xContent output in REST tests is fine for now, I'd just like to make sure we have both a roundtrip test (xContent/parsing) for the matching methods plus another unit test for the deprecated output also as a unit test later on. I think having unit tests for even this kind of simple things is important, REST tests are harder to run, maintain and debug which is why for easily testable things like xContent output I'd like to have unit tests in place. But sure this can be done in a follow up.